compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -183,6 +183,73 @@ TR::Instruction *loadConstant64(TR::CodeGenerator *cg, TR::Node *node, int64_t v
    return cursor;
    }
 
+   uint32_t getFPImm(double value)
+   {
+   uint32_t imm;
+   if (value == 0.5)
+      imm = 0x60;
+   else if (value == 1.0)
+      imm = 0x70;
+   else if (value == 2.0)
+      imm = 0x00;
+   else if (value == 4.0)
+      imm = 0x10;
+   else if (value == 8.0)
+      imm = 0x20;
+   else if (value == 16.0)
+      imm = 0x30;
+   else if (value == -0.5)
+      imm = 0xE0;
+   else if (value == -1.0)
+      imm = 0xF0;
+   else if (value == -2.0)
+      imm = 0x80;
+   else if (value == -4.0)
+      imm = 0x90;
+   else if (value == -8.0)
+      imm = 0xA0;
+   else if (value == -16.0)
+      imm = 0xB0;
+   else
+      imm = -1;
```

Comment:
Do not assign a negative value to a variable of an unsigned type.

Reply:
Thank you for your comment, I will change this to the `int32_t` type.